boot/cmdline.go

```diff
@@ -147,28 +147,28 @@ func composeCommandLine(model *asserts.Model, currentOrCandidate int, mode, syst
 
 // ComposeRecoveryCommandLine composes the kernel command line used when booting
 // a given system in recover mode.
-func ComposeRecoveryCommandLine(model *asserts.Model, system string) (string, error) {
-	return composeCommandLine(model, currentEdition, ModeRecover, system)
+func ComposeRecoveryCommandLine(model *asserts.Model, system, gadgetSnap string) (string, error) {
+	return composeCommandLine(model, currentEdition, ModeRecover, system, gadgetSnap)
 }
 
 // ComposeCommandLine composes the kernel command line used when booting the
 // system in run mode.
-func ComposeCommandLine(model *asserts.Model) (string, error) {
-	return composeCommandLine(model, currentEdition, ModeRun, "")
+func ComposeCommandLine(model *asserts.Model, gadgetDirOrSnap string) (string, error) {
```

Comment:
so if I understand we will sometimes pass a unpacked directory and someting a snap path? do we have other places like this?

naming wise:  gadgetUnpackDirOrSnapPath  perhaps

Reply:
Yes, for recovery systems we call ComposeRecoveryCommandLine which takes a path to gadget snap from seed. Other calls to Compose{Candidate}CommandLine use directory paths. In the end it all goes through the same helper.

Reply:
Went with gadgetDirOrSnapPath